webapp/platform/components/src/generic_modal/generic_modal.tsx

```diff
@@ -67,249 +68,242 @@ export type Props = {
      * in tests (Cypress, Playwright, etc.)
      */
     dataTestId?: string;
+
+    /**
+     * Delay in milliseconds before activating the focus trap.
+     *
+     * This is useful for modals with dynamic content that might not be fully
+     * rendered when the modal is opened. The delay allows the DOM to settle
+     * before the focus trap identifies focusable elements. ie. MultiSelect
+     */
+    focusTrapDelay?: number;
 };
 
-type State = {
-    show: boolean;
-    isFocalTrapActive: boolean;
-}
-export class GenericModal extends React.PureComponent<Props, State> {
-    static defaultProps: Partial<Props> = {
-        show: true,
-        id: 'genericModal',
-        autoCloseOnCancelButton: true,
-        autoCloseOnConfirmButton: true,
-        enforceFocus: true,
-        keyboardEscape: true,
-        bodyPadding: true,
-        showCloseButton: true,
-        showHeader: true,
-        modalLocation: 'center',
-    };
+export const GenericModal: React.FC<Props> = (props) => {
+    const {
+        show = true,
+        id = 'genericModal',
+        autoCloseOnCancelButton = true,
+        autoCloseOnConfirmButton = true,
+        enforceFocus = true,
+        keyboardEscape = true,
+        bodyPadding = true,
+        showCloseButton = true,
+        showHeader = true,
+        modalLocation = 'center',
+    } = props;
```

Comment:
If we are going to destructure the props, let's destructure them on the function argument.